### PR TITLE
Bug 1574628 - Improve graph colors and add symbols

### PR DESCRIPTION
**Pull request by yogmel**
## Description of issue
The Graph view in Perfherder is not very usable for color blind users, mainly when there are multiple types of data plotted.
To improve it, it was suggested to choose a friendlier **color palette** and also use auxiliary **symbols** to help distinguish between data.

- [x] Change symbols in graph to different ones
  - [X] When graph is first created
  - [x] When more graphs are added
- [x] Insert symbol in `LegendCard`
- [X] Propose better color palette

## Proposed Solution
- In order to change the graph symbols, the `VictoryScatter` prop [`symbol`](https://formidable.com/open-source/victory/docs/victory-scatter/#symbol) in VictoryChart's library was used. The symbol name is passed from `GraphView` to `GraphContainer`, based on the stored value from `constants.js`

- In `LegendCard`, the name of the symbols have been retrieved, but in order to insert the icon, some decisions must be made - the icons are `<svg>` and live in another file? Are they React Components? Maybe we could use the same svg Victory Chart uses.

- As for the color palette, some [research](https://venngage.com/blog/color-blind-friendly-palette/) and [experiments](https://coolors.co/ffe2f5-de9b14-f9ed00-e1dabd-abc798) were made. 3 of the 6 colors were changed.

## Testing
A visual test was made, with help of [NoCoffee Add-On](https://addons.mozilla.org/en-US/firefox/addon/nocoffee/). It was tested with all available Color deficiency options.

Screenshot of proposed color scheme for Perfherder's graph. There are legend cards on the left with 6 different border colors. On the right, there is the graph, with different symbols scattered

### CODE REVIEW

**Comment by sarah-clements** [comment]
Can you look into creating 2 more color palettes and posting screen shots? It'd be nice for users who are not color blind if the color palette were harmonious and complemented the colors used in the UI (dark grey and turquoise blue). This is the challenge I ran into when selecting the color palette that's currently used - how to make it dynamic and visually pleasing while being accessible to all :)

**Comment by yogmel** [comment]
Well, THAT'S a challenge!

I have put together 3 color palettes and their related color blindness screenshots, as NoCoffee shows. The following comment will have the screenshots of the graph for each palette.

------------

**Color Palette 1**

cp01

------------

**Color Palette 2**

cp02

--------------

**Color Palette 3**

cp03

**Comment by yogmel** [comment]
**Color Palette 1**


------

**Color Palette 2**


------

**Color Palette 3**

**Comment by sarah-clements** [comment]
hrm, I think the first palette is my preference even though the some of the colors look similar when simulated for different types of color blindness (although using the shapes for points will help). The second palette seems to be the best choice in that regard. 

However, I think we should get an opinion from people who use this app before merging it @davehunt and @ionutgoldan. Do you have a preference for one of these color palettes? (Note: Ionut won't be available until after the holidays)

**Comment by davehunt** [comment]
I prefer the first palette, but I agree that the second would be more accessible to colour blind users. I do find the variety of symbols distracting though. Could we get away with filled and non-filled versions of circles, squares, and triangles? That would give us six symbols using only three shapes.

**Comment by yogmel** [comment]
Thanks @davehunt for your input! I've made some tests with your suggestion. Screenshots are below:

**Color Palette 1**


---------------

**Color Palette 2**

**Comment by sarah-clements** [comment]
@davehunt Not sure if you're back from the holidays. Do you have time this week to look at Mellina's changes?

**Comment by davehunt** [comment]
> Not sure if you're back from the holidays. Do you have time this week to look at Mellina's changes?

I'm back. I think the latest screenshots look good but I don't want to be a blocker for this patch. Let's get @ionutgoldan's feedback.

**Comment by ionutgoldan** [comment]
I think we should go for Color Palette 2, mostly because data points can overlap one another & the higher contrast would help in distinguishing them.

**Comment by yogmel** [comment]
image

Some comments:

1. @sclements I know we talked about the Legend icons to be png files, but then I remembered that the icons **change colors**. That's why I have created `GraphIcon.jsx` component;

2. When turned off the graph, the default icon in the Legend Card is an outlined circle with a gray color.

3. There is a supposed _bug_ when choosing more than 6 graphs and toggling between them. If you turn off one of the first 6 graphs and turn on the 7th, for example, it doesn't appear in the Graph. I investigated, and it looks like the `stroke` color is not passed down to the new Graph.

I still have to take a look at it, as toggling on and off the graphs messes the color + symbol association between the Legend and Data Points. If you have insights for me, I would appreciate them.

**Comment by sarah-clements** [comment]
>     3. There is a supposed _bug_ when choosing more than 6 graphs and toggling between them. If you turn off one of the first 6 graphs and turn on the 7th, for example, it doesn't appear in the Graph. I investigated, and it looks like the `stroke` color is not passed down to the new Graph.

I'm working on a fix for that bug, so don't worry about it.

> I still have to take a look at it, as toggling on and off the graphs messes the color + symbol association between the Legend and Data Points. If you have insights for me, I would appreciate them.

If this is too complicated then I think we can leave the symbol off of the legend for now. As long as the colors are distinctive enough, the symbols I think are an "extra" to help with overlapped data points. However, from your screenshot I think two of the dark purples are too close in color. Can you swap one out for the green from the first color palette?

**Comment by yogmel** [comment]
@sarah-clements I've mirrored your solution and apparently it's working right.

I've tried using the green of the other palette, but it didn't have enough contrast for Protanopia color deficiency. I increased the brightness of the current hue:

image

**Comment by yogmel** [comment]
@sarah-clements here are some more color palettes, as we discussed yesterday. Please let me know your thoughts :) 

## Color Palette 4

ref01

------------

## Color Palette 5

ref02

--------------

## Color Palette 6

ref03

**Comment by sarah-clements** [comment]
Thanks for iterating on these @yogmel. Is the aqua/cyan color in palette 5 the same as the new darker-info color? I think it'd help to see these plotted.

**Comment by yogmel** [comment]
@sarah-clements sure. You're right, it feels different when we see the colors on the graphs.

## Color Palette 4


---------

## Color Palette 5



-------

## Color Palette 6

**Comment by sarah-clements** [comment]
I think let's go with the 2nd color palette. I completely missed the fact that you had added the symbols to the legend cards that correspond with the data points on the graph, so then having two colors that are close in color isn't too bad. My apologies :) These other palettes look great, but I see Ionut's point about having colors that look quite different from each other. Thanks for iterating on this!

**Comment by yogmel** [comment]
@sarah-clements no problem :) I have sent another update with the chosen color palette.

**Comment by yogmel** [comment]
Thanks, @camd and @sarah-clements for your support :)

File: ui/css/perf.css [hunk 1]
```diff
@@ -450,27 +450,27 @@ li.pagination-active.active > button {
 
 /* graph colors */
 .orange {
-  border-left-color: #ffba4a;
+  border-left-color: #ffb851;
 }
 
 .firered {
-  border-left-color: #c92b2a;
+  border-left-color: #c92d2f;
```
Comment (sarah-clements) [inline ui/css/perf.css]: For CSS names we use snake case (multiple words are separated with a hyphen so its easier to read).

File: ui/css/perf.css [hunk 2]
```diff
@@ -449,32 +449,28 @@ li.pagination-active.active > button {
...
 
-.scarlet {
-  border-left-color: #b81752;
+.blue-bell {
+  border-left-color: #464876;
 }
 
-.turquoise {
-  border-left-color: #17a2b8;
+.gras {
```
Comment (sarah-clements) [inline ui/css/perf.css]: Is this supposed to be "grass"? As in "grass-green"?

Comment (yogmel) [reply ui/css/perf.css]: image


I used the coolors.co name. I thought it was a good idea to abbreviate this one for some reason.
How about `purple`?

Comment (sarah-clements) [reply ui/css/perf.css]: 'gras' on its own doesn't really make sense so `purple` is fine :)

Comment (yogmel) [reply ui/css/perf.css]: Ok! Done!

**Comment by yogmel** [comment]
Great! No problem, it's a pleasure!

**Pull request merged**